playbooks/roles/hosts.update/tasks/centos.yml

```diff
@@ -1,12 +1,62 @@
 ---
+- name: Install custom kernel
+  when: config.nodes[inventory_hostname].kernel is defined
+  vars:
+    version: "{{ config.nodes[inventory_hostname].kernel.split('-')[0] }}"
+    release: "{{ config.nodes[inventory_hostname].kernel.split('-')[1] }}.el{{ config.os[config.nodes[inventory_hostname].os].version }}"
+    arch: "x86_64"
+  block:
+    - name: Install repository management tool
+      yum:
+        name: createrepo_c
+        state: latest
+
+    - name: Create local repository directory
+      file:
+        path: /root/repo
+        state: directory
+
+    - name: Download kernel packages
+      get_url:
+        dest: "/root/repo/kernel{{item}}-{{version}}-{{release}}.{{arch}}.rpm"
+        url: "https://kojihub.stream.centos.org/kojifiles/packages/kernel/{{version}}/{{release}}/{{arch}}/kernel{{item}}-{{version}}-{{release}}.{{arch}}.rpm"
+      loop:
+        - ""
+        - "-core"
+        - "-modules"
+        - "-modules-core"
+        - "-devel"
+        - "-headers"
+
+    - name: Create local repository
+      command: createrepo /root/repo
```

Comment:
I think we can go for a direct install of required (specific) kernel packages from url using dnf/yum without the creation of local repository.